metaflow/plugins/pypi/micromamba.py

```diff
@@ -0,0 +1,295 @@
+        #    index
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            env = {
+                "MAMBA_ADD_PIP_AS_PYTHON_DEPENDENCY": "true",
+                "CONDA_SUBDIR": platform,
+                # "CONDA_UNSATISFIABLE_HINTS_CHECK_DEPTH": "0" # https://github.com/conda/conda/issues/9862
+            }
+            cmd = [
+                "create",
+                "--yes",
+                "--quiet",
+                "--dry-run",
+                "--no-extra-safety-checks",
+                "--repodata-ttl=86400",
+                "--retry-clean-cache",
+                "--prefix=%s/prefix" % tmp_dir,
+            ]
+            # Introduce conda-forge as a default channel
+            for channel in self.info()["channels"] or ["conda-forge"]:
+                cmd.append("--channel=%s" % channel)
+
+            for package, version in packages.items():
+                cmd.append("%s==%s" % (package, version))
+            if python:
+                cmd.append("python==%s" % python)
+            # TODO: Ensure a human readable message is returned when the environment
+            #       can't be resolved for any and all reasons.
+            return [
+                {k: v for k, v in item.items() if k in ["url"]}
+                for item in self._call(cmd, env)["actions"]["LINK"]
+            ]
+
+    def download(self, id_, packages, python, platform):
+        # Unfortunately all the packages need to be catalogued in package cache
+        # because of which this function can't be parallelized
+
+        # Micromamba is painfully slow in determining if many packages are infact
+        # already cached. As a perf heuristic, we check if the environment already
+        # exists to short circuit package downloads.
+        if self.path_to_environment(id_, platform):
+            return
+
+        prefix = "{env_dirs}/{keyword}/{platform}/{id}".format(
+            env_dirs=self.info()["envs_dirs"][0],
+            platform=platform,
+            keyword="metaflow",  # indicates metaflow generated environment
+            id=id_,
+        )
+
+        # Another forced perf heuristic to skip cross-platform downloads.
+        if os.path.exists(f"{prefix}/fake.done"):
+            return
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            env = {
+                "CONDA_SUBDIR": platform,
+            }
+            cmd = [
+                "create",
+                "--yes",
+                "--no-deps",
+                "--download-only",
+                "--safety-checks=disabled",
+                "--no-extra-safety-checks",
+                "--repodata-ttl=86400",
+                "--prefix=%s/prefix" % tmp_dir,
+                "--quiet",
+            ]
+            for package in packages:
+                cmd.append("{url}".format(**package))
+
+            self._call(cmd, env)
+            # Perf optimization to skip cross-platform downloads.
```

Comment:
no - this is to skip cross-platform downloads since the heuristic above it only accounts for environments that are created - no environment is created locally for a cross-platform environment